### ISSUE TITLE
Vite ignoring resolve.mainFields in configuration

### ISSUE THREAD

**Issue by vembye** [electron-vite/vite-plugin-electron#217]
# Problem
Using the [default config](https://github.com/electron-vite/vite-plugin-electron/blob/6f50f421c973303c0d385336d847cd65a6c29d1c/src/utils.ts#L23), vite doesn't seem to respect the resolve.mainField.

We are using ws in our electron setup, and when using vite-plugin-electron vite kept importing the browser module from ws, corrupting the build. 

Seeing the correct approach in default config, ignoring the `browser` as a mainField. Unfortunately vite kept using the browser field. 

Turns out resolve.mainField is not respected when targeting browser (which is vite default).

# Solution

Enforce main entry build to be `ssr: true`.

I'm ready to create PR if issue is accepted.

**Comment by caoxiemeihao** [comment]
Hey! 👋
However, you can override the default configuration by configuring `electron({})` in `vite.cinfig.ts`.

**Comment by vembye** [comment]
Definitely, and how we have done it, so change not required to get it working. 

But debugging of multiple hours to figure out why the (already existing) clause of handling browser field didn't take effect, could have been avoided, and should for new-comers IMO 💡

It would have saved me a lot of frustrations at least 🙏🏻

**Comment by subframe7536** [comment]
Close for stale, feel free to reopen
